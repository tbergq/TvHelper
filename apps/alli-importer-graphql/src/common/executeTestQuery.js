// @flow

import { graphql } from 'graphql';

import schema from '../Schema';

const createContext = () => ({});

export default function executeTestQuery(
  query: string,
  variables: ?{ ... },
  request: ?{ ... } = {},
): Promise<{ ... }> {
  // $FlowExpectedError: Ok for testing purposes
  return graphql(schema, query, null, createContext(request), variables);
}
