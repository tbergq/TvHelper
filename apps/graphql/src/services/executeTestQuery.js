// @flow

import { graphql } from 'graphql';

import schema from '../Schema';
import createContext from './createGraphqlContext';

export default function executeTestQuery(
  query: string,
  variables: ?{ ... },
  request: ?{ ... } = {},
): Promise<{ ... }> {
  // $FlowExpectedError: Ok for testing purposes
  return graphql(schema, query, null, createContext(request), variables);
}
