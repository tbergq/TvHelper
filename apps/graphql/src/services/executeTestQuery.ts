import { graphql } from 'graphql';

import schema from '../Schema';
import createContext from './createGraphqlContext';

export default function executeTestQuery(
  query: string,
  variables: {} | null | undefined,
  request: {} | null | undefined = {},
): Promise<{}> {
  // @ts-ignore: Ok for testing purposes
  return graphql(schema, query, null, createContext(request), variables);
}
