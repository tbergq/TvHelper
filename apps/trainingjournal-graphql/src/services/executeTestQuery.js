// @flow

import { graphql } from 'graphql';

import schema from '../Schema';

export default function executeTestQuery(query: string, variables: ?{ ... }): Promise<{ ... }> {
  return graphql(schema, query, null, null, variables);
}
