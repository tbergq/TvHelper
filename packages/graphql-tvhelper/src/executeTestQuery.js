// @flow

import { graphql as originalGraphQL, GraphQLSchema, GraphQLObjectType } from 'graphql';

import SearchTvShow from './search/queries/SearchTvShow';

const RootQuery = new GraphQLObjectType({
  name: 'RootQuery',
  description: 'Root Query',
  fields: {
    searchTvShow: SearchTvShow,
  },
});

const schema = new GraphQLSchema({
  query: RootQuery,
});

/**
 * Executes GraphQL query using our schema. This function should
 * be used only during testing because it uses faked context.
 */
export default function executeTestQuery(
  query: string,
  variables: ?{ ... },
  context: ?{ ... },
): Promise<Object> {
  return originalGraphQL(schema, query, null, context, variables);
}
