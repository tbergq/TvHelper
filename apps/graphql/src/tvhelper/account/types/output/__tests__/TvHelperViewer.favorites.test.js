// @flow

import {
  UserRepository,
  FavoritesRepository,
  tvHelperConnection,
} from '@tbergq/tvhelper-persistence';

import executeTestQuery from '../../../../../services/executeTestQuery';

const query = `query FavoritesQuery {
  viewer {
    ... on TvHelperViewer{
      favorites {
        edges {
          node {
            id
            name
          }
        }
      }
    }
  }
}`;

let userId;

beforeEach(async () => {
  const user = await UserRepository.createUser({
    email: 'lol@lol.no',
    username: 'tito',
    password: 'a',
  });
  userId = user.id;
  await FavoritesRepository.createFavorite(userId, '6');
});

afterEach(async () => {
  await tvHelperConnection.collection('users').drop();
  await tvHelperConnection.collection('favorites').drop();
});

it('renders TJViewer type', async () => {
  const data = await executeTestQuery(query, {}, { user: { app: 'tvhelper', id: userId } });
  expect(data).toMatchSnapshot();
});
