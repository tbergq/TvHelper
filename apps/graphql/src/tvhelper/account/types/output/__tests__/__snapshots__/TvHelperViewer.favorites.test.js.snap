// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TJViewer type 1`] = `
Object {
  "data": Object {
    "viewer": Object {
      "favorites": Object {
        "edges": Array [
          Object {
            "node": Object {
              "id": "VHZTaG93OjY",
              "name": "The 100",
            },
          },
        ],
      },
    },
  },
}
`;
